controllers/spiaccesstokenbinding_controller.go

```diff
@@ -203,9 +204,30 @@ func (r *SPIAccessTokenBindingReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
+	expectedLifetimeDuration := r.Configuration.AccessTokenBindingTtl
+	infiniteDuration := false
+	if binding.Spec.Lifetime != "" {
```

Comment:
can we maybe instead of this another block of ifs have single function (with proper tests) like `func tokenLifetim(...) (*time.Duration, error)` that will take all this logic and return final duration of the binding or nil for infinite?

Reply:
well. if we don't afraid the one-time funcs and `nil` returns (oh gave me those optionals back), that probably looks little better.